### PR TITLE
SPINEDEM-1680 Undraft smoke tests

### PR DESCRIPTION
**Pull request by davidhamill1-nhs**
## Summary
Undraft smoke tests.

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [flake8]
 max-line-length=120
-exclude = .git,__pycache__,dist,node_modules/*,utils/*,tests/functional/test_env/*,sandbox/node_modules/*
+exclude = .git,__pycache__,dist,node_modules/*,utils/*,tests/functional/test_env/*,sandbox/node_modules/*,.venv/*
```
Comment (richardcoates5-nhs) [inline .flake8:3]: I assume this was t fix that issue with flatten?

Am I right in thinking none of our code goes in to the .venv folder? It's just there for packages we pull in?

Comment (davidhamill1-nhs) [reply .flake8:3]: Someone merged the `sandbox/node_modules/*` before me, which resolved the flatten flake issues.

`.venv/*` has appeared in my repo and was causing thousands of lines to be output. It doesn't sit in the repo, so no, none of our code goes in to there.

File: tests/user_restricted/test_user_restricted.py [hunk 2]
```diff
@@ -31,16 +33,30 @@ def test_retrieve_deprecated_url(self, headers_with_token):
...
+                                      nhsd_apim_config,
+                                      _test_app_credentials):
+        headers = helpers.get_headers(apigee_environment,
+                                      nhsd_apim_config,
+                                      _test_app_credentials,
+                                      AUTH_HEALTHCARE_WORKER)
+        self.retrieve_patient_and_assert(retrieve[0], headers)
 
+    @pytest.mark.nhsd_apim_authorization(AUTH_HEALTHCARE_WORKER)
+    def test_retrieve_patient_for_non_int(self, headers_with_token):
```
Comment (richardcoates5-nhs) [inline tests/user_restricted/test_user_restricted.py:50]: Are only smoke tests run against int?

File: .flake8 (hunk 1 above)
Comment (davidhamill1-nhs) [reply .flake8:3]: hence I have added it to the end. Not sure why I may be the only one being affected.

File: tests/user_restricted/test_user_restricted.py (hunk 2 above)
Comment (davidhamill1-nhs) [reply tests/user_restricted/test_user_restricted.py:50]: I'm not convinced I'll keep the skipif marker with later refactoring. But didn't want to make a big upheaval to only do it differently once I do the larger refactoring for this ticket.

Comment (ChrisClarke-NHSD) [inline tests/user_restricted/test_user_restricted.py:50]: Does this need marking?
```python
@pytest.mark.skipif(ENVIRONMENT == 'int')
```

Comment (davidhamill1-nhs) [reply tests/user_restricted/test_user_restricted.py:50]: Not necessarily. As the tests in INT are defined by the smoke_test marker,  they will never see this test: See https://github.com/NHSDigital/personal-demographics-service-api/blob/dbf7b91767dcfb301d796a9fd4c49809fd753505/azure/templates/pds-tests-int.yml#L49

Comment (ChrisClarke-NHSD) [reply tests/user_restricted/test_user_restricted.py:50]: Though actually, is this even a patient by this point? and is it used?

File: tests/user_restricted/test_user_restricted.py [hunk 3]
```diff
@@ -31,16 +33,30 @@ def test_retrieve_deprecated_url(self, headers_with_token):
...
+        self.retrieve_patient_and_assert(retrieve[0], self.headers)
+
+    def retrieve_patient_and_assert(self, paitent: dict, headers):
```
[the remark this answers is not in the document]
Comment (ChrisClarke-NHSD) [reply tests/user_restricted/test_user_restricted.py]: Though, actually, is this even used?

File: tests/user_restricted/test_user_restricted.py [hunk 4]
```diff
@@ -135,16 +151,28 @@ def test_retrieve_patient_with_missing_x_request_header(self, headers_with_token
 
 class TestUserRestrictedSearchPatient:
 
-    # @pytest.mark.smoke_test
     @pytest.mark.nhsd_apim_authorization(AUTH_HEALTHCARE_WORKER)
-    def test_search_patient_happy_path(self, headers_with_token):
+    def test_search_patient_happy_path_for_non_int(self, headers_with_token):
```
Comment (ChrisClarke-NHSD) [inline tests/user_restricted/test_user_restricted.py:155]: again, does this need marking so it doesn't run on int (I admit to some ignorance of exactly how pytest works - something i very much need to remedy)

Comment (davidhamill1-nhs) [reply tests/user_restricted/test_user_restricted.py:155]: See above ... https://github.com/NHSDigital/personal-demographics-service-api/pull/921#discussion_r1331735830

File: tests/user_restricted/test_user_restricted.py (hunk 2 above)
Comment (davidhamill1-nhs) [reply tests/user_restricted/test_user_restricted.py:50]: Agree, it's/they're a bit naff. But I think the refactoring steps to come in this ticket "Move the user-restricted tests under the functional/* group and refactor to use an appropriate Feature file" will see these test rewritten anyway, so don't really want to waste time cleaning up too much.

**Review by richardcoates5-nhs — APPROVED** [review]

File: tests/user_restricted/test_user_restricted.py (hunk 2 above)
Comment (ChrisClarke-NHSD) [reply tests/user_restricted/test_user_restricted.py:50]: Fair enough

**Review by ChrisClarke-NHSD — APPROVED** [review]

**Pull request merged**